test/svid/jwtsvid/test_jwt_svid.py

```diff
@@ -0,0 +1,160 @@
+        ),  # no aud
+        (
+            jwt.encode(
+                {
+                    'aud': ['spire'],
+                    'sub': 'spiffeid://somwhere.over.the',
+                },
+                'secret',
+                headers={'alg': 'ES384', 'typ': 'JWT'},
+            ),
+            ["spire"],
+            InvalidClaimError._text.format('exp'),
+        ),  # no exp
+        (
+            jwt.encode(
+                {
+                    'aud': ['spire'],
+                    'exp': timegm(
+                        (
+                            datetime.datetime.utcnow() - datetime.timedelta(hours=1)
+                        ).utctimetuple()
+                    ),
+                },
+                'secret',
+                headers={'alg': 'RS512', 'typ': 'JWT'},
+            ),
+            ["spire"],
+            InvalidClaimError._text.format('sub'),
+        ),  # no sub
+        (
+            jwt.encode(
+                {
+                    'aud': ['spire'],
+                    'sub': 'spiffeid://somwhere.over.the',
+                    'exp': timegm(
+                        (
+                            datetime.datetime.utcnow() - datetime.timedelta(hours=1)
+                        ).utctimetuple()
+                    ),
+                },
+                'secret',
+                headers={'alg': 'PS512', 'typ': 'JOSE'},
+            ),
+            ["spire"],
+            TokenExpiredError._text,
+        ),  # expired token
+    ],
+)
+def test_invalid_token_parse_insecure(test_input_token, test_input_audience, expected):
+    with pytest.raises(JwtSvidError) as exception:
+        JwtSvid.parse_insecure(test_input_token, test_input_audience)
+
+    assert str(exception.value) == expected
+
+
+@pytest.mark.parametrize(
+    'test_input_token,test_input_audience, expected',
+    [
+        (
+            jwt.encode(
+                {
+                    'aud': ['spire'],
+                    'sub': 'spiffe://test.org/',
+                    'exp': timegm(
+                        (
+                            datetime.datetime.utcnow() + datetime.timedelta(hours=100)
+                        ).utctimetuple()
+                    ),
+                },
+                'secret',
+                headers={'alg': 'RS256', 'typ': 'JWT'},
+            ),
+            ['spire'],
+            'spiffe://test.org/',
+        ),
+        (
+            jwt.encode(
+                {
+                    'aud': ['spire', 'test', 'valid'],
+                    'sub': 'spiffe://test.orgcom.br/',
+                    'exp': timegm(
+                        (
+                            datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+                        ).utctimetuple()
+                    ),
+                },
+                'secret key',
+                headers={'alg': 'PS384', 'typ': 'JOSE'},
+            ),
+            ['spire', 'test'],
+            "spiffe://test.orgcom.br/",
```

Comment:
would be useful to add a testcase using a hardcoded token fetched from the workload api, like this one?

```
ES256TOKEN =  'eyJhbGciOiJFUzI1NiIsImtpZCI6Imd1eTdsOWZSQzhkQW1IUmFtaFpQbktRa3lId2FHQzR0IiwidHlwIjoiSldUIn0.eyJhdWQiOlsib3RoZXItc2VydmljZSJdLCJleHAiOjE2MTIyOTAxODMsImlhdCI6MTYxMjI4OTg4Mywic3ViIjoic3BpZmZlOi8vZXhhbXBsZS5vcmcvc2VydmljZSJ9.W7CLQvYVBQ8Zg3ELcuB1K9hE4I9wyCMB_8PJTZXbjnlMBcgd0VDbSm5OjoqcGQF975eaVl_AdkryJ_lzxsEQ4A'
```

Reply:
I think so, I will do it.

Reply:
@maxlambrecht , as we chatted, SPIRE does not generated a long-very long lived token and after sometime, I will start getting TokenExpiredException when I was supposed to get a valid JwtSVID. I will put this test idea on hold for the future if we find a way around this. Thanks!